check_diff/src/lib.rs

```diff
@@ -270,3 +362,41 @@ pub fn compile_rustfmt(
         feature_runner,
     });
 }
+
+pub fn search_for_rs_files(repo: &Path) -> impl Iterator<Item = PathBuf> {
+    return WalkDir::new(repo).into_iter().filter_map(|e| match e.ok() {
+        Some(entry) => {
+            let path = entry.path();
+            if path.is_file() && path.extension().map_or(false, |ext| ext == "rs") {
+                return Some(entry.into_path());
+            }
+            return None;
+        }
+        None => None,
+    });
+}
+
+pub fn check_diff(config: Option<Vec<String>>, runners: CheckDiffRunners, repo: &Path) -> i32 {
```

Comment:
Would be nice to add a doc comment for `check_diff`, especially because it's marked as `pub`.